storops/unity/resource/lun.py

```diff
@@ -353,6 +358,43 @@ def thin_clone(self, name, io_limit_policy=None, description=None):
         return TCHelper.thin_clone(self._cli, self, name, io_limit_policy,
                                    description)
 
+    def _is_move_session_supported(self, dest):
+        if self.is_thin_clone:
+            log.error('Not support move session, source lun is thin clone.')
+            return False
+        if self.is_data_reduction_enabled and not dest.is_all_flash:
+            log.error('Not support move session, source lun is compressed, '
+                      'but destination pool is not all flash pool.')
+            return False
+        return True
+
+    def migrate(self, dest, **kwargs):
+        if not self._is_move_session_supported(dest):
+            return False
+
+        interval = kwargs.pop('interval', 5)
+        timeout = kwargs.pop('timeout', 1800)
+
+        @retryz.retry(timeout=timeout, wait=interval, on_return=False)
+        def _do_check_move_session(move_session_id):
+            move_session = clz.get(self._cli, _id=move_session_id)
+            if move_session.state == enums.MoveSessionStateEnum.COMPLETED:
+                return True
+            return False
+
+        clz = storops.unity.resource.move_session.UnityMoveSession
+        is_compressed = self.is_data_reduction_enabled
+
+        try:
+            move_session = clz.create(self._cli, self, dest,
```

Comment:
What if the lun is in cg?

Reply:
If lun in cg, REST will return 409 source or dest does not exist error, storops will raise UnityMigrationSourceDestNotExistsError exception.